### PR TITLE
CT-58 Membership Reference app extension

### PR DESCRIPTION
**Pull request by gabiayako**
This PR replaces the following app extensions
- `user-membership` - used in the field `members` of `workingGroups` in GP2 and `members` of `projects` in GP2
- `contributing-cohort-memberships` - used in the field `contributingCohorts` of `users` in GP2
- `user-team-memberships` - used in the field `teams` of `users` in CRN

by the new app extension called `membership-reference` that will be used in the fields listed above plus the field `members` of `workingGroups` in CRN.

Also it was created all migrations necessary to change the field editor of these fields listed above. 

---

Using the new `membership-reference` extension:

#### Members in Working Groups in CRN
members-wg-crn

#### Teams in Users in CRN
team-users-crn

#### Members in Working Groups in GP2
members-wg-gp2

#### Contributing Cohorts in Users in GP2
contributing-users-gp2

### CODE REVIEW

**Comment by gabiayako** [comment]
rebased to master to solve conflicts

**Comment by gabiayako** [comment]
rebased to master to solve conflicts

**Review by peterstarling — APPROVED** [review]

File: packages/contentful-app-extensions/membership-reference/src/locations/Field.tsx [hunk 1]
```diff
@@ -20,41 +20,72 @@ import {
...
-      </Heading>
-      <Paragraph marginBottom="none">{role}</Paragraph>
+      <Heading marginBottom="none">{getEntityDisplayName()}</Heading>
+      {entityName === 'user' &&
+        showUserEmail &&
+        data.fields.email?.['en-US'] && (
+          <Paragraph marginBottom="none">
+            {data.fields.email?.['en-US']}
+          </Paragraph>
+        )}
```
Comment (gabiayako) [inline packages/contentful-app-extensions/membership-reference/src/locations/Field.tsx:84]: @dianaguimaraes the issue was that I was trying to get the email from the membership entry before not the user entry 😛 thanks for the review 😄 

I've added this boolean parameter `showUserEmail` with default value `false` so we only need to add it to the fields where we want to show the email.
Screenshot 2023-07-26 at 13 56 39

So I added this param to the migrations `project` and `working group` below 

```
  const projects = migration.editContentType('projects');
  projects.changeFieldControl('members', 'app', 'Yp64pYYDuRNHdvAAAJPYa', {
    entityName: 'user',
    showUserEmail: true,     <------------
    showLinkEntityAction: false,
    showCreateEntityAction: true,
  });
};
```

```
  const workingGroups = migration.editContentType('workingGroups');
  workingGroups.changeFieldControl('members', 'app', 'Yp64pYYDuRNHdvAAAJPYa', {
    entityName: 'user',
    showUserEmail: true,     <------------
    showLinkEntityAction: false,
    showCreateEntityAction: true,
  });
```

**Comment by gabiayako** [comment]
rebased to master to solve conflicts

**Review by dianaguimaraes — APPROVED** [review]

**Comment by gabiayako** [comment]
rebasing to master before merge

**Pull request merged**